x-pack/plugins/monitoring/server/lib/alerts/fetch_cpu_usage_node_stats.ts

```diff
@@ -147,38 +342,52 @@ export async function fetchCpuUsageNodeStats(
     },
   };
 
-  try {
-    if (filterQuery) {
-      const filterQueryObject = JSON.parse(filterQuery);
-      params.body.query.bool.filter.push(filterQueryObject);
-    }
-  } catch (e) {
-    // meh
+  if (filterQuery) {
+    (params.query!.bool!.filter! as QueryDslQueryContainer[]).push(filterQuery);
   }
 
-  const response = await esClient.search(params);
-  const stats: AlertCpuUsageNodeStats[] = [];
-  const clusterBuckets = get(
-    response,
-    'aggregations.clusters.buckets',
-    []
-  ) as ClusterBucketESResponse[];
-  for (const clusterBucket of clusterBuckets) {
-    for (const node of clusterBucket.nodes.buckets) {
-      const lastBucket = get(node, 'histo.buckets[1]', {});
-      const indexName = get(node, 'index.buckets[0].key', '');
-      const stat = {
-        clusterUuid: clusterBucket.key,
-        nodeId: node.key,
-        nodeName: get(node, 'name.buckets[0].key'),
-        cpuUsage: get(node, 'average_cpu.value'),
-        containerUsage: get(lastBucket, 'usage_deriv.normalized_value'),
-        containerPeriods: get(lastBucket, 'periods_deriv.normalized_value'),
-        containerQuota: get(node, 'average_quota.value'),
-        ccs: indexName.includes(':') ? indexName.split(':')[0] : null,
-      };
-      stats.push(stat);
-    }
+  const response = (await esClient.search<unknown>(params)) as unknown as InferSearchResponseOf<
+    unknown,
+    typeof params
+  >;
+
+  if (!response.aggregations) {
+    throw new Error('Failed to resolve needed aggregations for CPU Usage Rule');
   }
-  return stats;
+
+  return response.aggregations.clusters.buckets.flatMap((cluster) => {
+    return cluster.nodes.buckets.map((node): AlertCpuUsageNodeStats => {
+      let nodeName;
+      if (node.name.buckets.length) {
+        nodeName = node.name.buckets[0].key as string;
+      }
+
+      let ccs;
+      if (node.index.buckets.length) {
+        const index = node.index.buckets[0].key as string;
+        ccs = index.includes(':') ? index.split(':')[0] : undefined;
+      }
+
+      const runningInAContainer =
+        node.quota_micros_min.value !== null || node.quota_micros_max.value !== null;
+      if (runningInAContainer) {
+        return {
+          clusterUuid: cluster.key as string,
+          nodeId: node.key as string,
+          cpuUsage: node.average_cpu.value ?? undefined,
+          nodeName,
+          ccs,
+          unexpectedLimits: true,
+        };
+      }
+
+      return {
+        clusterUuid: cluster.key as string,
+        nodeId: node.key as string,
+        cpuUsage: node.average_cpu.value ?? undefined,
+        nodeName,
+        ccs,
+      };
```

Comment:
**suggestion(non-blocking)**: I was looking for differences on these returns but I didn't find anything apart the `unexpectedLimits` flag, I personally find it more readable having a single return statement in this case such as:
```js
const runningInAContainer = node.quota_micros_min.value !== null || node.quota_micros_max.value !== null;

return {
  clusterUuid: cluster.key as string,
  nodeId: node.key as string,
  cpuUsage: node.average_cpu.value ?? undefined,
  nodeName,
  ccs,
  unexpectedLimits: runningInAContainer
};
```
but it is just a readability preference, feel free to ignore this one :)

Reply:
Makes sense, I was sticking to the same format as the container path where it makes more sense with the different returns (since the data returned doesn't overlap). I'll join these!